tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_reconfigure.c

```diff
@@ -0,0 +1,89 @@
+/* main_reconfigure.c - Application main entry point */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#include <bluetooth/gatt.h>
+
+CREATE_FLAG(flag_reconfigured);
+
+void att_mtu_updated(struct bt_conn *conn, uint16_t tx, uint16_t rx)
+{
+	printk("MTU Updated: tx %d, rx %d\n", tx, rx);
+	SET_FLAG(flag_reconfigured);
+}
+
+static struct bt_gatt_cb cb = {
+	.att_mtu_updated = att_mtu_updated,
+};
+
+static void test_peripheral_main(void)
+{
+	peripheral_setup_and_connect();
+
+	while (bt_eatt_count(default_conn) < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(10));
+	}
+
+	bt_gatt_cb_register(&cb);
+
+	WAIT_FOR_FLAG(flag_reconfigured);
+
+	disconnect();
+
+	PASS("EATT Peripheral tests Passed\n");
+}
+
+#define NEW_MTU 100
+
+static void test_central_main(void)
+{
+	int err;
+
+	central_setup_and_connect();
+
+	while (bt_eatt_count(default_conn) < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(10));
+	}
+
+	bt_gatt_cb_register(&cb);
+
+	err = bt_eatt_reconfigure(default_conn, NEW_MTU);
```

Comment:
We should consider disallowing the sending of a reconfigure request with an MTU larger than `BT_L2CAP_RX_MTU`. Here, we are lying to the peer by saying that we can receive a packet that will not fit in the ACL buffer size.

Reply:
Is it a problem that the MTU is larger than the ACL buffer? 

in l2cap.c we have
```c
	req->mtu = sys_cpu_to_le16(mtu);

	/* MPS shall not be bigger than MTU + BT_L2CAP_SDU_HDR_SIZE
	 * as the remaining bytes cannot be used.
	 */
	req->mps = sys_cpu_to_le16(MIN(mtu + BT_L2CAP_SDU_HDR_SIZE,
				       BT_L2CAP_RX_MTU));
```

where we restrict the MPS to be less than or equal ot the RX_MTU - Do we need to restrict the `req->mtu`?

Reply:
I was concerned about the peer sending too much data because the MTU is larger than the buffer size. If the MPS is small enough, it should be ok.